apps/backend/src/routes/socket/index.ts

```diff
@@ -0,0 +1,120 @@
+import { Server, Socket } from "socket.io";
+import { server } from "@/index";
+import {
+  SocketRequestType,
+  SocketRequestSchema,
+  SocketRequest,
+  SocketResponseType,
+  SocketResponse,
+  SocketErrorPayload,
+  MapRequestToResponse,
+  errorToString
+} from "@types";
+import { Controller } from "@/lib/controller";
+
+export const wsServer = new Server(server);
+
+const controller = new Controller();
+
+const clientsSockets: Record<string, Socket> = {};
+
+const handleError = (socket: Socket, recipient: string, error: string): void => {
+  const payload: SocketErrorPayload = {
+    error,
+  }
+
+  const resp: SocketResponse = {
+    type: SocketResponseType.ERROR,
+    recipientSigPubKey: recipient,
+    senderSigPubKey: "",
+    payload
+  };
+
+  socket.send(JSON.stringify(resp))
+  return;
+}
+
+wsServer.on('connection', (socket: Socket) => {
+  socket.on('open', () => {
```

Comment:
not urgent, but does socket.io have authenticated connections? so auth token doesnt need to be sent with every message?

Reply:
addressed [de28bf9](https://github.com/cursive-team/connections/pull/191/commits/de28bf9aee23d2bbaba9ddcc10b8f97649e14d06)

Commit message: 
```
a commit of beauty: in providing the auth token, the user can be looked up and the pubkey-to-socket lookup entry can be set up. This allows the CONNECT-CONNECT_SUCCESS step to be removed as an established connection already implies that lookup has been set up. This also allows the sender to be removed from all future requests, as the information has been attached to the user's associated socket. Glorious.
```